src/VisualStudio/Core/Def/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -55,11 +52,13 @@ internal interface IWorkspaceProjectContext : IDisposable
         void RemoveAnalyzerReference(string referencePath);
 
         // Files.
-        void AddSourceFile(string filePath, bool isInCurrentContext = true, IEnumerable<string> folderNames = null, SourceCodeKind sourceCodeKind = SourceCodeKind.Regular);
+        void AddSourceFile(string filePath, bool isInCurrentContext = true, IEnumerable<string>? folderNames = null, SourceCodeKind sourceCodeKind = SourceCodeKind.Regular);
         void RemoveSourceFile(string filePath);
+        [Obsolete($"Call the {nameof(AddAdditionalFile)} method that takes folder names.")]
         void AddAdditionalFile(string filePath, bool isInCurrentContext = true);
+        void AddAdditionalFile(string filePath, IEnumerable<string> folderNames, bool isInCurrentContext = true);
```

Comment:
The other folderNames are nullable but not this one. Why?

Reply:
If that was also made nullable and optional, then a call of AddAdditionalFile("File.txt") was ambiguous between the two calls, since there's no special logic for the compiler to ignore Obsolete methods in that case. Conveniently the compiler [is thinking about that](https://github.com/dotnet/csharplang/pull/7707) so that's good. :smile: